runtime/serde/common/src/aws/smithy/kotlin/runtime/serde/SdkFieldDescriptor.kt

```diff
@@ -6,23 +6,6 @@ package aws.smithy.kotlin.runtime.serde
 
 import aws.smithy.kotlin.runtime.InternalApi
 
-/**
- * This tag interface provides a mechanism to attach type-specific metadata to any field.
- * See [aws.smithy.kotlin.runtime.serde.xml.XmlList] for an example implementation.
- *
- * For example, to specify that a list should be serialized in XML such that values are wrapped
- * in a tag called "boo", pass an instance of XmlList to the FieldDescriptor of `XmlList(elementName="boo")`.
- */
-@InternalApi
-public interface FieldTrait
```

Comment:
Why are these moved?

Reply:
Since the file was named `SdkFieldDescriptor` I thought it would make more sense to have field traits in a different place, I can change it to how it was though